app/src/components/utils.ts

```diff
@@ -68,6 +70,39 @@ export function getGitRevision(): string {
   }
 }
 
+/**
+ * @function getCurrentIdentity
+ * Attempts to acquire current identity value.
+ * Always takes first non-default value available. Yields `defaultValue` otherwise.
+ * @param {object} currentUser The express request currentUser object
+ * @param {string} [defaultValue=undefined] An optional default return value
+ * @returns {string} The current user identifier if applicable, or `defaultValue`
+ */
+export function getCurrentIdentity(currentUser: CurrentUser | undefined, defaultValue: string | undefined = undefined) {
+  return parseIdentityKeyClaims()
```

Comment:
i always thought this was overly complicated in COMS and i didnt like all the NIL's and defaultValue stuff.
could be one function 
```js
getCurrentUserId(currentUser){
// get claim name matching config.identityKey
// get claim value from currentUser.tokenPayload
// do database query to return userId
}
```
that being said.. if this works maybe it's better for now.

Reply:
I agree. Might try to clean it up during the refactoring process before going to prod.